packages/fxa-auth-server/scripts/cleanup-partial-firestore-customers.ts

```diff
@@ -0,0 +1,231 @@
+    this.accountDeleteManager = Container.get(AccountDeleteManager);
+    this.accountManager = Container.get(AccountManager);
+    this.cloudTaskQueue = new PQueue({ concurrency: rateLimit });
+  }
+
+  /**
+   * Fetches subscriptions from Firestore paginated by batchSize
+   * @param startAfter ID of the last element of the previous batch for pagination
+   * @returns A list of subscriptions from firestore
+   */
+  private async fetchFirestoreCustomersUid(
+    startAfter: QueryDocumentSnapshot | undefined
+  ) {
+    this.log.debug('CleanupFirestoreHelper.fetchFirestoreCustomersUid.start', {
+      startAfterDocId: startAfter?.id,
+    });
+
+    const collectionPrefix = `${this.config.authFirestore.prefix}stripe-`;
+    const customerCollection = `${collectionPrefix}customers`;
+    const customerSnap = startAfter
+      ? await this.firestore
+          .collectionGroup(customerCollection)
+          .orderBy(FieldPath.documentId())
+          .startAfter(startAfter)
+          .limit(this.batchSize)
+          .get()
+      : await this.firestore
+          .collectionGroup(customerCollection)
+          .orderBy(FieldPath.documentId())
+          .limit(this.batchSize)
+          .get();
+
+    // Customer document ids are the FxA Account UID
+    const uids = customerSnap.docs.map((doc) => doc.id);
+    const newStartAfter = customerSnap.docs.at(-1);
+
+    return { uids, startAfter: newStartAfter };
+  }
+
+  private async getUidsForDelete(uids: string[]) {
+    this.log.debug('CleanupFirestoreHelper.getUidsForDelete.start', {
+      nrOfUids: uids.length,
+    });
+
+    // Fetch all accounts for uids
+    const accounts = await this.accountManager.getAccounts(uids);
+    const accountUids = accounts.map((account) =>
+      uuidTransformer.from(account.uid)
+    );
+    // Filter uids where account does not exist
+    // Uids where account doesn't exist, should be deleted
+    const uidsForDelete = uids.filter(
+      (uid) => !accountUids.find((accountId) => accountId === uid)
+    );
+
+    return uidsForDelete;
+  }
+
+  private async callPartialDelete(uids: string[]) {
+    this.log.debug('CleanupFirestoreHelper.callPartialDelete.start', {
+      uidsForDelete: uids.length,
+    });
+    if (this.dryRun) {
+      this.log.debug('CleanupFirestoreHelper.callPartialDelete.skipped', {
+        dryRun: this.dryRun,
+        uidNumbers: uids.length,
+      });
+
+      return;
+    }
+
+    await this.cloudTaskQueue.addAll(
+      uids.map(
+        (uid) => () =>
+          this.accountDeleteManager.enqueue({
+            uid,
+            reason: AccountDeleteReasonsMap.cleanup,
+          })
+      )
+    );
+  }
+
+  async cleanup() {
+    this.log.debug('DeletePartial.start', {
+      batchSize: this.batchSize,
+      dryRun: this.dryRun,
+    });
+    let startAfter: QueryDocumentSnapshot | undefined = undefined;
+    let batchCounter = 0;
+    const hasMore = true;
+    while (hasMore) {
```

Comment:
Where does this get updated?

Reply:
I think it :yolo:'s out on L151's `break` statement.

Reply:
Yup I preferred the `break` to exit the while loop.

Eslint wasn't all to happy with `while (true) {...` (no-constant-condition), so I just put it in a `const hasMore = true` on the line above.

I suppose the right move would have been to ignore the eslint rule.